mantidproject/mantid
# Load Dialog crash after closing when loading

**Description of work.**
This PR fixes a crash caused by closing the LoadDialog while it was still attempting to load a file.

The code was previously using a while loop to wait for the LoadDialog to search for the specified file before continuing with accepting the dialog. In this while loop it was calling `QApplication->processEvents()` to refresh the GUI. This means you can close the LoadDialog whilst it is still in the while loop, as events are being processed. This is not recommended in the Qt documentation for processEvents https://doc.qt.io/qt-6/qcoreapplication.html#processEvents .

A safer way to handle this is to connect a signal to the FileFinder worker and then do any post-processing in a separate slot function. This means if the LoadDialog is closed while it is still searching for a file, the signal connection is also removed, and this avoids the read access violation.

This has also allowed me to do some cleaning and remove some unnecessary code.

**To test:**
Follow the instructions in the attached issue
Also try load files which should be found successfully

Fixes #35056

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
